src/Setup_geometry.jl

```diff
@@ -9,14 +9,121 @@ using GeoParams
+    ```julia
+    julia> Grid = ReadLaMEM_InputFile("test_files/SaltModels.dat")
+    LaMEM Grid:
+      nel         : (32, 32, 32)
+      marker/cell : (3, 3, 3)
+      markers     : (96, 96, 96)
+      x           ϵ [-3.0 : 3.0]
+      y           ϵ [-2.0 : 2.0]
+      z           ϵ [-2.0 : 0.0]
+    julia> Phases = zeros(Int32,   size(Grid.X));
+    julia> Temp   = zeros(Float64, size(Grid.X));
+    julia> AddBox!(Phases,Temp,Grid, xlim=(0,500), zlim=(-50,0), phase=ConstantPhase(3), DipAngle=10, T=ConstantTemp(1000))
+    julia> addStripes!(Phases, Grid, stripAxes=(1,1,1), stripeWidth=0.2, stripeSpacing=1, Origin=nothing, StrikeAngle=0, DipAngle=10, phase=ConstantPhase(3), stripePhase=ConstantPhase(4))
+    julia> Model3D = ParaviewData(Grid, (Phases=Phases,Temp=Temp)); # Create Cartesian model
+    julia> Write_Paraview(Model3D,"LaMEM_ModelSetup")           # Save model to paraview
+    1-element Vector{String}:
+     "LaMEM_ModelSetup.vts"
+    ```
+"""
+function addStripes!(Phase, Grid::AbstractGeneralGrid;                # required input
+    stripAxes       = (1,1,0),                          # activate stripes along dimensions x, y and z when set to 1
+    stripeWidth     =  0.2,                             # full width of a stripe
+    stripeSpacing   =  1,                               # spacing between two stripes centers
+    Origin          =  nothing,                         # origin
+    StrikeAngle     =  0,                               # strike
+    DipAngle        =  0,                               # dip angle
+    phase           =  ConstantPhase(3),                # phase to be striped
+    stripePhase     =  ConstantPhase(4))                # stripe phase
+
+    # warnings
+    if stripeWidth >= stripeSpacing/2.0
+        print("WARNING: stripeWidth should be strictly < stripeSpacing/2.0, otherwise phase is overwritten by the stripePhase\n")
+    elseif sum(stripAxes .== 0) == 3
+        print("WARNING: at least one axis should be set to 1 e.g. stripAxes = (1,0,0), otherwise no stripes will be added\n")
+    end
+
+    # Retrieve 3D data arrays for the grid
+    X,Y,Z = coordinate_grids(Grid)
+
+    # sets origin
+    if isnothing(Origin)
+        Origin = (maximum(X), maximum(Y), maximum(Z))  # upper-left corner
+    end
+
+    # Perform rotation of 3D coordinates:
+    Xrot = X .- Origin[1];
+    Yrot = Y .- Origin[2];
+    Zrot = Z .- Origin[3];
+
+    Rot3D!(Xrot,Yrot,Zrot, StrikeAngle, DipAngle)
+
+    ph_ind  = findall(Phase .== phase.phase);
+
+    ind = []
+    if stripAxes[1] == 1
+        indX     = findall( abs.(Xrot[ph_ind] .% stripeSpacing) .<= stripeWidth/2.0);
+        ind = vcat(ind,indX);
+    end
+    if stripAxes[2] == 1
+        indY     = findall( abs.(Yrot[ph_ind] .% stripeSpacing) .<= stripeWidth/2.0);
+        ind = vcat(ind,indY);
+    end
+    if stripAxes[3] == 1
+        indZ     = findall( abs.(Zrot[ph_ind] .% stripeSpacing) .<= stripeWidth/2.0);
+        ind = vcat(ind,indZ);
+    end
```

Comment:
You mean for the concatenation?
Can we do that in the case not all the dimensions are active? Here the point is that we only concatenate when the dimension(s) is defined in the stripAxes. What is concatenated depends on the user input. But maybe I misunderstand

Reply:
Ok you are right, I tried to make convert it in a loop and it gets actually much longer. Leave it as it is. Fixing `ind=Int64[]` was more important

Reply:
But in this case, you dont really need to use `findall`. You can just do
```julia
ph_ind = Phase .== phase.phase

if stripAxes[1] == 1
    indX     = abs.(Xrot[ph_ind] .% stripeSpacing) .<= stripeWidth/2
    ind = vcat(ind,indX);
end
if stripAxes[2] == 1
    indY     = abs.(Yrot[ph_ind] .% stripeSpacing) .<= stripeWidth/2
    ind = vcat(ind,indY);
end
if stripAxes[3] == 1
    indZ     = abs.(Zrot[ph_ind] .% stripeSpacing) .<= stripeWidth/2
    ind = vcat(ind,indZ);
end
```

Reply:
If I try the way you suggest it throws an error. Could that be because I work with a subset?

`Phase[ph_ind[ind]] .= stripePhase.phase;`

ERROR: BoundsError: attempt to access 96×96×96 BitArray{3} at index [[0, 0, 0, 0, 0, 0, 0, 0, 0, 0  …  0, 0, 0, 0, 0, 0, 0, 0, 0, 0]]
Stacktrace:
 [1] throw_boundserror(A::BitArray{3}, I::Tuple{Vector{Int64}})
   @ Base ./abstractarray.jl:734
 [2] checkbounds
   @ Base ./abstractarray.jl:699 [inlined]
 [3] _getindex
   @ Base ./multidimensional.jl:888 [inlined]
 [4] getindex(A::BitArray{3}, I::Vector{Int64})
   @ Base ./abstractarray.jl:1288
 [5] addStripes!(Phase::Array{…}, Grid::LaMEM_grid; stripAxes::Tuple{…}, stripeWidth::Float64, stripeSpacing::Int64, Origin::Nothing, StrikeAngle::Int64, DipAngle::Int64, phase::ConstantPhase, stripePhase::ConstantPhase)
   @ GeophysicalModelGenerator ~/GeophysicalModelGenerator.jl/src/Setup_geometry.jl:120
 [6] top-level scope
   @ REPL[7]:1
Some type information was truncated. Use `show(err)` to see complete types.

Reply:
oh yes because e.g. `Xrot[ph_ind]` is already a smaller array... Ok leave it as it was :)

Reply:
Ok! thanks for the review!